### PR TITLE
Fix #3753 comment count in retro summary

### LINKED ISSUE #3753: Tweak to make retro comments easier to find from summary
## Issue - Enhancement

Several users are still writing in worried that their retro output was "lost". The root trouble seems to be that they are conflating discussion comments with tasks, not realizing the distinction between the two.

Let's make some quick tweaks to the summary format to continue to reduce this tension.

### Acceptance Criteria (optional)
Summary updated where users can:
   - See a cell added to the left of tasks showing `n comments`

image

   - Template string updated for Topic header to `{TopicName} – {Comments.count()? "See Comments({Comments.count()}) : "No Comments"`

image

- **Estimated effort:** 2 hours

### PR DESCRIPTION
**Pull request by ackernaut**
Fixes #3753 

### Tests
- [ ] The title, votes, and comment link are all clickable; clicking takes you to the discussion
- [ ] Depending on count, the comment link label is:
  - No Comments (neutral gray link instead of blue)
  - See 1 Comment
  - See N Comments
  - See 100+ Comments (I’m not sure we need to query for many comments so I capped it at 100.)
- [ ] This works in both the email and browser views
- [ ] The demo summary has static topic titles and votes (no links back to the comments)

### Note
Doesn’t solve for bugs (re)discovered that exist in prod:
- The email topic links are busted with utm params before the `/discuss/N` sub path of the URL, see #3779
- The timeline shows meetings that were ended early, the summary link doesn’t work, see #3778

### CODE REVIEW

**Comment by ackernaut** [comment]
Here’s how the first pass is styled (note the title, votes, and comment link are all clickable):

Screen Shot 2020-04-21 at 5 25 24 PM

**Comment by ackernaut** [comment]
Hmm, I noticed the links in the email version (opposed to browser view) don’t link directly to the topic. Was that intentional @mattkrick ?

File: packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx [hunk 1]
```diff
@@ -81,6 +109,13 @@ export default createFragmentContainer(RetroTopic, {
     fragment RetroTopic_topic on RetroReflectionGroup {
       title
       voteCount
+      thread(first: 101) @connection(key: "DiscussionThread_thread") {
```
Comment (mattkrick) [inline packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx:112]: ooo, this is gonna get pretty expensive for the DB, maybe instead we just have a "commentCount" field? it returns the `count` on the comment table indexed on the thread id.

this fixes the thread/comment ambiguity. for example with threads, if we say "see 10 comments" and all 10 are tasks, there won't be any comments.

File: packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx [hunk 2]
```diff
@@ -41,17 +53,26 @@ interface Props {
...
+      ? 'No Comments'
+      : commentLength >= 101
+      ? 'See 100+ Comments'
+      : `See ${commentLength} ${plural(commentLength, 'Comment')}`
   return (
     <>
       <tr>
         <td align='center' style={{paddingTop: 20}}>
           <AnchorIfEmail href={to} isEmail={isEmail} style={topicThemeHeading}>
+            {'“'}
```
Comment (mattkrick) [inline packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx:73]: IMO it was a little cleaner without quotes (i think we got rid of them in a previous revision?)

File: packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx [hunk 3]
```diff
@@ -41,17 +53,26 @@ interface Props {
...
-  const {reflections, title, voteCount} = topic
+  const {reflections, title, voteCount, thread} = topic
   const imageSource = isEmail ? 'static' : 'local'
   const icon = imageSource === 'local' ? 'thumb_up_18.svg' : 'thumb_up_18@3x.png'
   const src = `${emailDir}${icon}`
   const grid = useEmailItemGrid(reflections, 3)
+  const commentLength = thread.edges.length
+  const commentLinkLabel =
+    commentLength === 0
+      ? 'No Comments'
```
Comment (mattkrick) [inline packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx:64]: if there are no comments, I'd just hide it.

File: packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx [hunk 4]
```diff
@@ -41,17 +53,26 @@ interface Props {
...
+      : commentLength >= 101
+      ? 'See 100+ Comments'
```
Comment (mattkrick) [inline packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx:66]: nice thinking!

File: packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx (hunk 1 above)
Comment (ackernaut) [reply packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopic.tsx:112]: @mattkrick perfect. I’m looking at adding a `commentCount` field to the type using `await dataLoader.get('commentsByThreadId').load(reflectionGroupId)` but now that I think about it that may be expensive, too? Instead should I add a commentCount field to the database record for the group and update that count with mutations?

**Comment by ackernaut** [comment]
@mattkrick one thing I noticed here and I’ve reproduced on staging and production. If I end a retro early, say in the group phase, a timeline event is made. Clicking on ‘review a summary’ for that event fails silently for the user and creates an error on the server:
```
SEND TO SENTRY GraphQLError: Cannot return null for non-nullable field RetrospectiveMeeting.(reflectionGroups,reflectionGroups).
    at RetrospectiveMeetingreflectionGroupsMapItemHandler (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/src/execution.ts:237:9), <anonymous>:473:73)
    at RetrospectiveMeetingreflectionGroupsMapHandler (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/src/execution.ts:237:9), <anonymous>:751:8)
    at Object.safeMap (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/src/execution.ts:1045:5)
    at reflectionGroupsRetrospectiveMeetingreflectionGroupsResolverHandler (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/src/execution.ts:237:9), <anonymous>:771:13)
    at __parent.newMeeting.__value.then.__context.nullErrors.push.__context.GraphQLError.line (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/src/execution.ts:237:9), <anonymous>:3010:9)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) undefined
```

**Comment by ackernaut** [comment]
Another update here. I have a reliable comment count for real meetings. Now I’m looking at handling this for the demo summary.

**Comment by ackernaut** [comment]
There are 3 existing quirks/bugs that this doesn’t solve for (they can be see in prod):
1. In the email, the topic links don’t link directly to the topic but redirect (I think) to the last completed topic. In the browser, the link is direct which is a better UX IMO
2. In the demo, the topic links don’t work
3. Event cards are being created on the timeline for meetings that are ended early. Tapping ‘review a summary’ doesn’t work

**Comment by ackernaut** [comment]
Re: the topic links not working as intended in the email

The way the topic link is constructed for the email gives us the meeting URL, followed by the utm params, then the discuss path. It ends up looking something like:

```
http://localhost:3000/meet/Qoxhm2m4Fn?utm_source=summary+email&utm_medium=email&utm_campaign=after-meeting/discuss/1
```

Which of course doesn’t work as intended. I’m looking to use a helper function to create that link in the proper way, and we’re discussing whether or not those topic links benefit from including utm params.

**Comment by ackernaut** [comment]
@mattkrick okay, ready for a second review (updated tests). I’m going to create an issue for borked email topic links and timelines events for meetings that were ended early.

**Comment by mattkrick** [comment]
working on this now, just needs a dataloader & it'll be good to go!

**Comment by mattkrick** [comment]
hey neat, github lets you put things back in draft now! (not that this needs to be in draft, but i was just curious)

**Comment by mattkrick** [comment]
while adding the dataloader i noticed V8 had a mem leak in development. the cause was `options.stats` was being serialized on compile, which is just useless since the stats aren't even used. the fix was to fork webpack-hot-loader. there are still low hanging fruit if i fork webpack-hmr-loader, but that's fine

**Pull request merged**